test/functional/runner/index.js

```diff
@@ -391,10 +391,22 @@ function isTransactionCommand(command) {
   return ['startTransaction', 'commitTransaction', 'abortTransaction'].indexOf(command) !== -1;
 }
 
-function isTestRunnerCommand(command) {
-  return (
-    ['targetedFailPoint', 'assertSessionPinned', 'assertSessionUnpinned'].indexOf(command) !== -1
-  );
+function isTestRunnerCommand(context, commandName) {
+  const testRunnerContext = context.testRunner;
+
+  let methods = new Set();
+  let object = testRunnerContext;
+  do {
```

Comment:
nit: can we do it without a `do-while-break`? It seems like you could do this with a regular while loop:

```js
let object = testRunnerContext;
while(object !== Object.prototype) {
  ...
  object = Object.getPrototypeOf(object);
}
```

Reply:
I think its sort of precisely what a `do-while` is for: I want to do something with the value before iterating. This could be written a few different ways, but is there a more pressing concern about the way its written?

Reply:
I find the code confusing, and I'd like to avoid confusing code being introduced. Unrolling this, `Object.getPrototypeOf(object)` will never return a falsey value, so it is a misuse of what the `while` condition is for. This is effectively like writing:
```js
do {
  if (object === Object.prototype) break;
  
  ...
  object = Object.getPrototypeOf(object);
} while (true);
```

which is just a more convoluted way of writing:

```js
while(object !== Object.prototype) {
  ...
  object = Object.getPrototypeOf(object);
}
```

Reply:
done